src/nwp_consumer/internal/inputs/icon/client.py

```diff
@@ -0,0 +1,279 @@
+"""Implements a client to fetch ICON data from DWD."""
+import bz2
+import datetime as dt
+import pathlib
+import re
+import typing
+import urllib.request
+
+import requests
+import structlog
+import xarray as xr
+
+from nwp_consumer import internal
+
+from ._models import IconFileInfo
+
+log = structlog.getLogger()
+
+# See https://d-nb.info/1081305452/34 for a list of ICON parameters
+PARAMETER_RENAME_MAP: dict[str, str] = {
+    "t_2m": internal.OCFShortName.TemperatureAGL.value,
+    "clch": internal.OCFShortName.HighCloudCover.value,
+    "clcm": internal.OCFShortName.MediumCloudCover.value,
+    "clcl": internal.OCFShortName.LowCloudCover.value,
+    "asob_s": internal.OCFShortName.DownwardShortWaveRadiationFlux.value,
+    "athb_s": internal.OCFShortName.DownwardLongWaveRadiationFlux.value,
+    "w_snow": internal.OCFShortName.SnowDepthWaterEquivalent.value,
+    "relhum_2m": internal.OCFShortName.RelativeHumidityAGL.value,
+    "u_10m": internal.OCFShortName.WindUComponentAGL.value,
+    "v_10m": internal.OCFShortName.WindVComponentAGL.value,
+    "clat": "lat", # Icon has a seperate dataset for latitude...
+    "clon": "lon", # ... and longitude (for the global model)! Go figure
```

Comment:
do you have a link to any documention for this? Or perhaps it sin the link above

Reply:
Yes, I think it is mentioned in the link above.